src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -74,4 +110,52 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
     protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest request) {
         return AuthorisationUtil.getTokenPermissions(request);
     }
+
+    protected boolean isRequestRoeUpdate(HttpServletRequest request) {
+        final Map<String, List<String>> privileges = getERICTokenPermissions(request);
+        if (privileges.containsKey(COMPANY_NUMBER_KEY)) {
+            companyNumberInScope = privileges.get(COMPANY_NUMBER_KEY).get(0);
+            return StringUtils.isNotEmpty(companyNumberInScope) && companyNumberInScope.startsWith("OE");
+            }
```

Comment:
} needs aligning

Reply:
Should we error if it does not start with OE?

Reply:
I think interceptor will fail (return false) in this scenario so the remote call will ultimately error, but this should be proven by a unit-test - @bantao-ch can you confirm this please?

Adding a helpful log error message in this scenario might be useful though.

Reply:
Yes @AndreKramer7654 @mwestacott -- this will return false and fail ...

Reply:
Did you decide not to add the log error message @bantao-ch or was this missed?

Reply:
I've added the company_in_scope to the AuthMap instead would that suffice ?

Reply:
This is now added in the new revised logic ---

Reply:
👍 thanks.